### PR TITLE
Automatically detect virtual environments when used via `python -m uv`

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/uv/issues/1501

### CODE REVIEW

**Comment by zanieb** [comment]
cc @altendky :)

File: python/uv/__main__.py [hunk 1]
```diff
@@ -3,6 +3,26 @@
...
+    Find the virtual environment path for the current Python executable.
+    """
+
+    # If it's already set, then just use it
+    value = os.getenv("VIRTUAL_ENV")
+    if value:
+        return value
+
+    # Otherwise, check if we're in a venv
+    maybe_venv_dir = os.path.dirname(os.path.dirname(sys.executable))
```
Comment (henryiii) [inline python/uv/__main__.py:17]: Shouldn't this be something like `(Path(sys.prefix) / "pyvenv.cfg").exists()` (probably without the pathlib for speed, but I'm lazy here)? This assumes that there is exactly one level between the `bin` or `Scripts` dir and the base prefix, which I don't think is guaranteed.

Comment (zanieb) [reply python/uv/__main__.py:17]: Thanks! I thought what I was doing might be brittle.. let me look into that.

**Comment by altendky** [comment]
Thanks!

```console
$ python3.11 -m venv venv
```

```console
$ venv/bin/python -m pip install 'uv @ git+https://github.com/astral-sh/uv@1ed6ba0ba0d3756312085ec21c90469985b5bbb6'
Collecting uv@ git+https://github.com/astral-sh/uv@1ed6ba0ba0d3756312085ec21c90469985b5bbb6
  Using cached uv-0.1.2-py3-none-linux_x86_64.whl
Installing collected packages: uv
Successfully installed uv-0.1.2

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: /home/altendky/tmp/venv/bin/python -m pip install --upgrade pip
```

Installed `main` above and see it failing below.

```console
$ venv/bin/python -m uv pip install attrs
error: Failed to locate a virtualenv or Conda environment (checked: `VIRTUAL_ENV`, `CONDA_PREFIX`, and `.venv`). Run `uv venv` to create a virtualenv.
```

```console
$ venv/bin/python -m pip install 'uv @ git+https://github.com/astral-sh/uv@967188a914df3461b92de27e9b9234fd8c22be85'
Collecting uv@ git+https://github.com/astral-sh/uv@967188a914df3461b92de27e9b9234fd8c22be85
  Using cached uv-0.1.1-py3-none-linux_x86_64.whl
Installing collected packages: uv
  Attempting uninstall: uv
    Found existing installation: uv 0.1.2
    Uninstalling uv-0.1.2:
      Successfully uninstalled uv-0.1.2
Successfully installed uv-0.1.1

[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: /home/altendky/tmp/venv/bin/python -m pip install --upgrade pip
```

Installed this PR above and see it succeeding below.

```console
$ venv/bin/python -m uv pip install attrs
Resolved 1 package in 107ms
Installed 1 package in 9ms
 + attrs==23.2.0
```

File: python/uv/__main__.py (hunk 1 above)
Comment (altendky) [reply python/uv/__main__.py:17]: https://docs.python.org/3/library/sysconfig.html#installation-paths Looks like you can sometimes get a `posix_prefix`.  Might lead to more details about 'correctness' as well.

**Pull request merged**